users/helauth/clients.py

```diff
@@ -44,24 +49,219 @@ class HelsinkiProfileClient(BaseExternalServiceClient):
+        """
+        Refresh helsinki profile token.
+
+        :return: The new access token if refresh was successful, None otherwise.
+        """
+        user: AnyUser = request.user
+        if user.is_anonymous:
+            return None
+
+        access_token = TunnistamoClient.get_token(request)
+        if access_token is None:
+            msg = "Unable to get tunnistamo access token to refresh profile token."
+            logger.info(msg)
+            return None
+
+        url = f"{settings.TUNNISTAMO_BASE_URL}/api-tokens/"
+        headers: dict[str, str] = {"Authorization": f"Bearer {access_token}"}
+
+        response = cls.post(url=url, headers=headers)
+
+        if response.status_code != 200:
+            msg = (
+                f"Unable to get API tokens for helsinki profile for user {int(request.user.pk)}: "
+                f"[{response.status_code}] {response.text}"
+            )
+            logger.info(msg)
+            return None
+
+        session_data = get_session_data(request)
+        session_data["api_tokens"] = BaseExternalServiceClient.response_json(response=response)
+        return session_data["api_tokens"].get(settings.OPEN_CITY_PROFILE_SCOPE)
+
+    @classmethod
+    def graphql_request(cls, request: WSGIRequest, *, query: str) -> MyProfileData | None:
         token = cls.get_token(request)
         if token is None:
             return None
 
         url: str = settings.OPEN_CITY_PROFILE_GRAPHQL_API
-        request_data = {"query": cls._reservation_prefill_query()}
+        request_data = {"query": query}
         headers = {"Authorization": f"Bearer {token}"}
 
         response = cls.post(url=url, json=request_data, headers=headers)
         response_data = cls.response_json(response)
 
-        my_profile_data: MyProfileData = get_nested(response_data, "data", "myProfile", default={})
-        return ReservationPrefillParser(my_profile_data).parse()
+        return get_nested(response_data, "data", "myProfile", default={})
 
     @classmethod
-    def _reservation_prefill_query(cls) -> str:
-        primary_email_fragment = """
+    def response_json(cls, response: Response) -> dict[str, Any]:
+        if response.status_code != 200:
+            raise ExternalServiceRequestError(response, cls.SERVICE_NAME)
+
+        response_data = super().response_json(response)
+
+        errors: dict[str, Any] | None = response_data.get("errors")
+        if errors is not None:
+            msg = f"{cls.SERVICE_NAME.capitalize()}: Response contains errors."
+            SentryLogger.log_message(message=msg, details=errors, level="error")
+            raise ExternalServiceError(msg, details=errors)
+
+        return response_data
+
+    @classproperty
+    def reservation_prefill_query(cls) -> str:
+        return (
+            """
+            query {
+                myProfile {
+                    firstName
+                    lastName
+                    ...MyPrimaryEmail
```

Comment:
Nitpicking but could remove (or rename) the `My` prefix from the fragment names now that they can also be used to query another person's profile.

Reply:
Makes sense, I'll remove them.